addons/udes_stock/utils.py

```diff
@@ -0,0 +1,117 @@
+from urllib.parse import urljoin
+from odoo import _
+
+BASE_PRODUCT_IMAGE_URL = "/web/image/product.product/%i"
+
+
+def location_to_dict(location, **extras):
+    """Represent a location in a format that json can understand"""
+    # TODO: call from udes_simple_location_blocking and update with blocked location
+    # if location.u_blocked:
+    #     reason = location.u_blocked_reason or "no reason specified"
+    # else:
+    #     reason = False
+
+    loc = {
+        "barcode": location.barcode,
+        "name": location.name,
+        "id": location.id,
+        # "blocked": {"is_blocked": location.u_blocked, "reason": reason},
+        # TODO: enable when u_storage_format is implemented
+        # "storage_format": location.u_storage_format,
+    }
+    loc.update(extras)
+    return loc
+
+
+def product_to_dict(product, **extras):
+    """Represent a product in a format that json can understand"""
+    prod = {"name": product.name, "barcode": product.barcode, "display_name": product.display_name}
+    prod.update(extras)
+    return prod
+
+
+def product_quantity_label(product, quantity):
+    """A function for a common output format for product quantity labels"""
+    return "{name} x {quantity}".format(name=product.display_name, quantity=quantity)
+
+
+def package_product_quantity_label(package, product, quantity):
+    """A function for a common output format for package and product quantity labels"""
+    return "{package} {product_quantity}".format(
+        package=package.name, product_quantity=product_quantity_label(product, quantity)
+    )
+
+
+def format_picking_data_for_display_list_component(picking, order=True):
+    """Formats the data of a picking to be displayed in a list component
+
+    The data will be returned as a list of strings "packagename productname x productquantity"
+    """
+    contents = []
+    for package, package_lines in picking.move_line_ids.groupby("package_id", order):
+        for product, product_lines in package_lines.groupby("product_id", order):
+            contents.append(
+                package_product_quantity_label(
+                    package, product, int(sum(product_lines.mapped("product_uom_qty")))
+                )
+            )
+    return contents
+
+
+def batch_to_dict(batch):
```

Comment:
I think this method can be removed and used _get_info instead which is defined in udes_get_info module.

Reply:
Yes I think it would be best if it's removed or adapted, the only thing I'm worried about is the field "`"name": pick.batch_id.u_original_name or pick.batch_id.name` which __get_info wouldn't exactly replicate the same way. Happy to remove or expand to use __get_info.
This function is used by workflows.

Reply:
Can be put into extras i think and on location just add batch_id or in _get_info_field_names.
Would be good to use get_info module for this kind of dict converts

Reply:
As per discussion with Miquel, this will be removed and when workflows are ported a better way will be implemented to get this info. Most surely using __get_info() and not having a utils function just for this.